piet-cairo/src/blurred_rect.rs

```diff
@@ -0,0 +1,57 @@
+//! Helpers for efficiently drawing blurred rectangles.
+//!
+//! This function is not cairo-specific, but currently cairo is the only back-end
+//! that requires it, as other back-ends have their own implementation.
+//!
+//! As a future performance optimization, we might implement a cache of the computed
+//! images.
+
+use cairo::{Format, ImageSurface};
+
+use piet::kurbo::{Point, Rect};
+
+/// Extent to which to expand the blur.
+const BLUR_EXTENT: f64 = 2.5;
+
+pub fn compute_blurred_rect(rect: Rect, radius: f64) -> (ImageSurface, Point) {
+    let radius_recip = radius.recip();
+    let xmax = rect.width() * radius_recip;
+    let ymax = rect.height() * radius_recip;
+    let padding = BLUR_EXTENT * radius;
+    let rect_padded = rect.inflate(padding, padding);
+    let rect_exp = rect_padded.expand();
+    let xfrac = rect_padded.x0 - rect_exp.x0;
+    let yfrac = rect_padded.y0 - rect_exp.y0;
+    let width = rect_exp.width() as usize;
+    let height = rect_exp.height() as usize;
+    let strip = (0..width)
+        .map(|i| {
+            let x = ((i as f64) - (xfrac + padding)) * radius_recip;
+            (255.0 * 0.25) * (compute_erf7(x) + compute_erf7(xmax - x))
+        })
+        .collect::<Vec<_>>();
+    // TODO: don't panic on error
```

Comment:
Hard to know what to do here. if we return an error are we going to handle it any better? If this fails it's likely we're out of memory. I wish I had better guidelines for how to handle these sorts of errors. At a higher level, What's the best behaviour? Do we just skip this operation, but keep the process running?